labgrid/driver/power/gude8210.py

```diff
@@ -0,0 +1,77 @@
+from ..exception import ExecutionError
+from ...util.snmp import SimpleSNMP
+
+# Base OID for the Gude power switch as per the MIB file from gude8210
+POWER_OID_BASE = "1.3.6.1.4.1.28507.1.1.2.2.1.3"
+NUMBER_OF_OUTLETS = 8  # Max number of outlets from the MIB
+
+
+def power_set(host, port, index, value):
+    """
+    Sets the power state of a specific outlet on the Gude power switch.
+
+    Args:
+        host (str): The IP address of the power switch.
+        port (int or None): SNMP port, default is 161 or None.
+        index (int): Outlet index.
+        value (bool): True to turn on, False to turn off.
+        community (str): SNMP community string.
+    """
+    # Validate index within allowable range
+    if index is None or not (1 <= int(index) <= NUMBER_OF_OUTLETS):
+        raise ExecutionError("Invalid outlet index. Ensure the index is within the range 1 to 8.")
+
+    # Setup SNMP connection and OID for the target outlet
+    _snmp = SimpleSNMP(host, 'private', port=port)
+    oid = f"{POWER_OID_BASE}.{index}"
+    snmp_value = "1" if value else "0"  # SNMP value for on/off
+
+    print(f"Debug: Attempting SNMP SET on host {host}, OID {oid}, value {snmp_value}")
```

Comment:
Please switch this over to `logging.debug`, this should not be visible on non-debug usage. Same for the other prints below.

Reply:
Updated